plugin/storage/cassandra/spanstore/writer.go

```diff
@@ -130,8 +164,42 @@ func (s *SpanWriter) WriteSpan(span *model.Span) error {
 	return nil
 }
 
-func (s *SpanWriter) shouldIndexTag(tag dbmodel.TagInsertion) bool {
+func (s *SpanWriter) indexByDuration(span *dbmodel.Span) error {
+	query := s.session.Query(durationIndex)
+	timeBucket := int((span.StartTime / (60 * 60)) / 1000000) // 1hr in microseconds TODO use config
+	var err error
+	indexByOperationName := func(operationName string) {
+		q1 := query.Bind(span.Process.ServiceName, operationName, timeBucket, span.Duration, span.StartTime, span.TraceID)
+		if err2 := s.writerMetrics.durationIndex.Exec(q1, s.logger); err2 != nil {
+			err = err2
+		}
+	}
+	indexByOperationName("")                 // index by service name alone
+	indexByOperationName(span.OperationName) // index by service name and operation name
+	return err
+}
+
+func (s *SpanWriter) indexBySericeAndOperation(traceID model.TraceID, span *dbmodel.Span) error {
+	startTime := span.StartTime
+	bucketNo := traceID.Low % defaultNumBuckets
```

Comment:
if you want to do that, we need to have proof of uniform distribution of bucket numbers, e.g. as a metric. 

Why can't we simply use `atomic.AddUint32(&bucket, 1) % numBuckets` ?  I remember xys guys added consistent hash on trace ID, but I don't think this is relevant now.